src/main/java/application/SessionController.java

```diff
@@ -142,16 +114,62 @@ public GetInstanceResponseBean getInstance(@RequestBody String body) throws IOEx
-            return new EvaluateXPathResponseBean(formEntrySession, evaluateXPathRequestBean.getXpath());
-        } catch(Exception e){
-            e.printStackTrace();
-            return null;
-        }
+    public EvaluateXPathResponseBean evaluateXpath(@RequestBody EvaluateXPathRequestBean evaluateXPathRequestBean) throws Exception {
+        SerializableSession serializableSession = sessionRepo.find(evaluateXPathRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+        return new EvaluateXPathResponseBean(formEntrySession, evaluateXPathRequestBean.getXpath());
+    }
+
+    @RequestMapping(value = "/new_repeat", method = RequestMethod.GET)
+    @ResponseBody
+    public RepeatResponseBean newRepeat(@RequestBody RepeatRequestBean newRepeatRequestBean) throws Exception {
+        SerializableSession serializableSession = sessionRepo.find(newRepeatRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+
+        AnswerQuestionJson.descendRepeatToJson(formEntrySession.getFormEntryController(),
+                formEntrySession.getFormEntryModel(),
+                newRepeatRequestBean.getFormIndex());
+
+        serializableSession.setFormXml(formEntrySession.getFormXml());
+        serializableSession.setInstanceXml(formEntrySession.getInstanceXml());
+        sessionRepo.save(serializableSession);
+        JSONObject response =  AnswerQuestionJson.getCurrentJson(formEntrySession.getFormEntryController(),
+                formEntrySession.getFormEntryModel());
+        return mapper.readValue(response.toString(), RepeatResponseBean.class);
+    }
+
+    @RequestMapping(value = "/delete_repeat", method = RequestMethod.GET)
+    @ResponseBody
+    public RepeatResponseBean delete_repeat(@RequestBody RepeatRequestBean repeatRequestBean) throws Exception {
+        SerializableSession serializableSession = sessionRepo.find(repeatRequestBean.getSessionId());
+        FormEntrySession formEntrySession = new FormEntrySession(serializableSession);
+
+        JSONObject resp = AnswerQuestionJson.deleteRepeatToJson(formEntrySession.getFormEntryController(),
+                formEntrySession.getFormEntryModel(),
+                repeatRequestBean.getFormIndex());
+
+        serializableSession.setFormXml(formEntrySession.getFormXml());
+        serializableSession.setInstanceXml(formEntrySession.getInstanceXml());
+        sessionRepo.save(serializableSession);
+
+        JSONObject response =  AnswerQuestionJson.getCurrentJson(formEntrySession.getFormEntryController(),
+                formEntrySession.getFormEntryModel());
+
+        return mapper.readValue(response.toString(), RepeatResponseBean.class);
+    }
+
+    @RequestMapping("/filter_cases")
+    public CaseFilterResponseBean filterCasesHQ(@RequestBody CaseFilterRequestBean filterRequest) throws Exception {
```

Comment:
is this for backwards compatibility? or do we still need to filter cases sent from HQ?